<a href="https://colab.research.google.com/github/SEC-API-io/sec-api-cookbook/blob/main/notebooks/extractor-api/extract-sections-8-k.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Extract Item Sections from 8-K Filings

Extracting sections from 8-K filings is straightforward and can be accomplished with just a few lines of Python code. The `filing_url` parameter accepts both `.htm` and `.txt` versions of a filing. A complete list of supported item codes is available in the Request & Response section of the documentation.

The example below demonstrates how to extract Item 1.01, "Entry into a Material Definitive Agreement," as cleaned text:

In [ ]:
from sec_api import ExtractorApi

extractorApi = ExtractorApi("YOUR_API_KEY")

filing_url = "https://www.sec.gov/Archives/edgar/data/66600/000149315222016468/form8-k.htm"

# extract section 1.01 "Entry into Material Definitive Agreement" as cleaned text
section_text = extractorApi.get_section(filing_url, "1-1", "text")

print("Section 1.01 text:")
print("------------------")
print(section_text)

Section 1.01 text:
------------------
 Item 1.01 Entry into a Material Definitive Agreement. 

&#160; 

Quad M Solutions, Inc., an Idaho corporation, (the &#8220;Company&#8221; or &#8220;Quad M&#8221;), is a public holding company that offers staffing services and employee benefits, such as health plans, HR-human resources, and payroll services, to small and mid-sized group employers. The Company is filing this Current Report on Form 8-K to disclose recent material events, including the Company&#8217;s entry into a material agreements, through its wholly-owned subsidiary Physicians HealthCare Services LLC (&#8220;PHCS&#8221;), with Advent Health, a Florida-based clinically-integrated network that contracts with health care providers to provide certain Covered Services to Members (&#8220;Advent Health Participating Providers&#8221;) and has the ability to sign Payor contracts with Advent Health Participating Providers. 

&#160; 

Through PHCS, the Company now has immediate access to app

SEC filings might contain HTML character entities, such as `&#160;`, `&#8220;` and `&amp;`, and others, representing reserved characters, such as non-breaking space ` `, left double quotation mark `“`, and ampersand `&`, respectively. These entities are encoded in the HTML file, which can be converted to plain text using the `html.unescape` function from the `html` module. The `html.unescape` function converts HTML entities to their corresponding characters. For example, `&#160;` is converted to a non-breaking space, `&#8220;` is converted to a left double quotation mark `“`, and `&amp;` is converted to an ampersand `&`.

In [ ]:
import html

text = html.unescape(section_text)

print("Section 1.01 text after unescaping HTML character entities:")
print("-----------------------------------------------------------")
print(text.strip())

Section 1.01 text after unescaping HTML character entities:
-----------------------------------------------------------
Item 1.01 Entry into a Material Definitive Agreement. 

  

Quad M Solutions, Inc., an Idaho corporation, (the “Company” or “Quad M”), is a public holding company that offers staffing services and employee benefits, such as health plans, HR-human resources, and payroll services, to small and mid-sized group employers. The Company is filing this Current Report on Form 8-K to disclose recent material events, including the Company’s entry into a material agreements, through its wholly-owned subsidiary Physicians HealthCare Services LLC (“PHCS”), with Advent Health, a Florida-based clinically-integrated network that contracts with health care providers to provide certain Covered Services to Members (“Advent Health Participating Providers”) and has the ability to sign Payor contracts with Advent Health Participating Providers. 

  

Through PHCS, the Company now has immedi